src/selection.coffee

```diff
@@ -183,7 +183,7 @@ class Selection extends Model
 
   # Public: Clears the selection, moving the marker to the head.
   clear: ->
-    @marker.setProperties(goalBufferRange: null)
+    @marker.setProperties(goalBufferRange: null, goalScreenRange: null)
```

Comment:
That's totally reasonable. The only method using it is `merge`, which can be refactored to work directly with screen ranges as well. I'll explain this refactoring a bit more thoroughly in a comment later on :+1: